easybuild/easyblocks/w/wxpython.py

```diff
@@ -41,41 +43,106 @@ class EB_wxPython(PythonPackage):
 
     def build_step(self):
-        """No separate build step for wxPython."""
-        pass
+        """Custom build step for wxPython."""
+        if LooseVersion(self.version) >= LooseVersion("4"):
+            prebuild_opts = self.cfg['prebuildopts']
+            script = 'build.py'
+            wxflag = ''
+            BUILD_CMD = "%(prebuild_opts)s %(pycmd)s %(script)s --prefix=%(prefix)s -v"
+            base_cmd = BUILD_CMD % {
+                'prebuild_opts': prebuild_opts,
+                'pycmd': self.python_cmd,
+                'script': script,
+                'prefix': self.installdir,
+            }
+            if get_software_root('wxWidgets'):
+                wxflag = '--use_syswx'
+            else:
+                cmd = base_cmd + " build_wx"
+                run_cmd(cmd, log_all=True, simple=True)
+
+            cmd = base_cmd + " %s build_py" % wxflag
+            run_cmd(cmd, log_all=True, simple=True)
 
     def install_step(self):
-        """Custom install procedure for wxPython, using provided build-wxpython.py script."""
+        """Custom install procedure for wxPython."""
         # wxPython configure, build, and install with one script
-        script = os.path.join('wxPython', 'build-wxpython.py')
-        cmd = "{0} {1} --prefix={2} --wxpy_installdir={2} --install".format(self.python_cmd, script, self.installdir)
+        preinst_opts = self.cfg['preinstallopts']
+        INSTALL_CMD = "%(preinst_opts)s %(pycmd)s %(script)s --prefix=%(prefix)s"
+        if LooseVersion(self.version) >= LooseVersion("4"):
+            wxflag = ''
+            if get_software_root('wxWidgets'):
+                wxflag = '--use_syswx'
+            script = 'build.py'
+            cmd = INSTALL_CMD % {
+                'preinst_opts': preinst_opts,
+                'pycmd': self.python_cmd,
+                'script': script,
+                'prefix': self.installdir,
+            }
+            cmd = cmd + " %s -v install" % wxflag
+        else:
+            script = os.path.join('wxPython', 'build-wxpython.py')
+            cmd = INSTALL_CMD % {
+                'preinst_opts': preinst_opts,
+                'pycmd': self.python_cmd,
+                'script': script,
+                'prefix': self.installdir,
+            }
+            cmd = cmd + " --wxpy_installdir=%s --install" % self.installdir
+
         run_cmd(cmd, log_all=True, simple=True)
 
     def sanity_check_step(self):
         """Custom sanity check for wxPython."""
         majver = '.'.join(self.version.split('.')[:2])
         shlib_ext = get_shared_lib_ext()
-        py_bins = ['alacarte', 'alamode', 'crust', 'shell', 'wrap', 'wxrc']
+        py_bins = ['crust', 'shell', 'wxrc']
+        files = []
+        dirs = []
+        if LooseVersion(self.version) < LooseVersion("4"):
+            files.extend([os.path.join('bin', 'wxrc')])
+            dirs.extend(['include', 'share'])
+            py_bins.extend(['alacarte', 'alamode', 'wrap'])
```

Comment:
why not add `wxrc` here too?

Reply:
Ask the person who wrote the ver<4 part.... but probably because there is a wxrc binary and not just a pywxrc? And it is already in py_bins a couple of lines higher up.